### DISCUSSION TITLE
Am I doing something wrong?

### DISCUSSION THREAD

**Discussion by thepragmaticmero** [casey/just#2390] [Q&A] [answered]
I just want to use the just file from my parent directory.
My shell is here: `/home/user/kitty/cat`
```
$ pwd
/home/user/kitty/cat
```
I want to execute the just file located at `/home/user/kitty/justfile`
If I use this command I get an error 
```
$ just -d ../ -f justfile
error: Failed to read justfile at '/home/user/kitty/cat/justfile': No such file or directory (os error 2)
```
If I type the whole path still get the same error
```
$ just -d /home/user/kitty -f justfile
error: Failed to read justfile at '/home/user/kitty/cat/justfile': No such file or directory (os error 2)
```
I'm explicitly telling the command "search inside this directory this justfile called 'justfile' ". Why is it searching inside the "cat" directory? I'm really confused.

**Comment by thepragmaticmero** [comment, marked as answer]
Ok I found the answer.
```
$ just -d .. -f ../justfile
```
I'm wondering why separate -d and -f? Genuine question huh.

**Reply by laniakea64** [reply]
> ```
> $ just -d .. -f ../justfile
> ```
> 
> I'm wondering why separate -d and -f?

`-d` sets/overrides the working directory for running the recipes in the justfile (which otherwise is specified in the justfile, or defaults to the directory containing the justfile).  It's not used for locating the justfile.

**Reply by casey** [reply]
Also, if the justfile is in parent directory, you should be able to invoke recipes from it without needing `-d` or `-f`.

**Reply by thepragmaticmero** [reply]
woah, learning something new everyday I think, that's an amazing feature and so obvious that my brain didn't recognized it. Good program casey~ I'll stick with just :D

**Reply by casey** [reply]
Thanks! Funnily enough, it's actually the first feature that `just` had that distinguished it from `make`. It always irritated me that I had to be in the root directory to run `make` refcipes.
